apps/demos/src/app/demos/submodules/dyn-forms/components/simple/simple.form.ts

```diff
@@ -99,9 +99,21 @@ export function simpleForm(
           }),
           createMatConfig('INPUT', {
             name: 'zipCode',
-            options: { validators: { required: null, minLength: 4 } },
+            options: {
+              matchers: [
+                {
+                  matcher: 'ENABLE',
+                  negate: true,
+                  operator: 'AND',
+                  when: [
+                    { path: 'firstName', value: 'Mateo' },
+                    { path: 'country', value: 'CO' },
+                  ]
+                },
+              ]
+            },
```

Comment:
@Yumitoya8569 what do you think of this?
I've managed to have a working logic-engine here.
The `matcher` will receive the result of the AND/OR conditions reactively (the DEFAULT condition wires the control.valueChanges and the matcher will combine their results)
Both the `matcher` and the `conditions` are able to `negate` the result to cause the opposite effect.

We can provide custom `matcher` handlers to provide any feature inside the FormTreeNode
and custom `conditions` to trigger custom or complex cases.

I will wire the `VISIBLE` | `INVISIBLE` | `HIDDEN` matchers to modify the styles of the `dyn-factory` next.

Reply:
The user may need to be prompted for the default value of  `operator`.

BTW i noticed `matcher` change to `matchers` in current version, very useful features!

Reply:
> i noticed matcher change to matchers in current version, very useful features!

yeah, I thought that we could want to run more than 1 matcher for a set of conditions :)

> The user may need to be prompted for the default value of operator.

it doesn't depend exclusively on the data in the form?
the conditions will trigger on each change of the target `paths`
so a default vlaue could be overriden when the form is initialized.

Reply:
@matheo
My bad, i mean `operator` is optional but didn't get what the default value is.
I trace it until find the code `config.operator === 'OR' ? results.some(Boolean) : results.every(Boolean)`.

Reply:
ah I get the confusion now!
`DEFAULT` is the id of the condition provided by the library to process the path and value.
You could provide new conditions too for custom or async logic :)

And yeah, once any Condition trigger a result (because of valueChange or statusChange or any)
the `operator` will need all the conditions true (for `AND`) and only one (with`OR`).